torch/distributed/_tensor/device_mesh.py

```diff
@@ -106,6 +106,7 @@ def __init__(
         mesh: MeshExprT,
         *,
         _init_process_groups: bool = True,
+        _sync_seed: bool = True,
```

Comment:
We shouldn't add an argument for device mesh initialization here. We should broadcast the seed in `get_or_create_default_pg` from rank0, then directly call the API in `random.py`(with the seed in device mesh) in the dispatcher instead.